### PR TITLE
First pass at pagination

### PR DESCRIPTION
**Pull request by duglin**
Just a first draft of a "pagination" spec. Rather than put it into Discovery, I thought that we might be able to define it once and reuse it for all of the specs.

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by duglin** [comment]
on 7/16 call @clemensv recommended we look at RFC7232 - section 3.3

**Comment by duglin** [comment]
> s/sever/server/

@drewpca fixed!

File: pagination.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+The follow list of attributes MAY be included in a client's request for
+a set of records from a server. The client MUST only specify these on
+the initial request for set. If the results can not fit within one
+message then these attributes MUST NOT be added to subsequent requests
+by the client. Note, the server MAY include these attributes within
+the URI-reference returned in a response message, but the client MUST NOT
+modify those values.
+
+#### size
```
Comment (sdatspun2) [inline pagination.md:33]: suggestion: limit (size) and offset (page) are more commonly used in pagination queries

Comment (duglin) [reply pagination.md:33]: ok done

**Comment by duglin** [comment]
I reworked the 'expires' stuff to leverage the HTTP "Expires" header. Most of the rest of the doc remains the same (just s/size/limit/ and s/page/offset/) but that's just a syntax change, no semantic change.

**Comment by duglin** [comment]
All - sorry for not pushing the latest before the call.  I just updated it and made it clear that the use of things like "offset" in the URLs is an impl choice. The spec doesn't mandate how the server does its chunking.

**Comment by duglin** [comment]
Approved on 8/13 call

**Pull request merged**